lib/Jsrt/Jsrt.cpp

```diff
@@ -5112,6 +5112,71 @@ CHAKRA_API JsCreatePromise(_Out_ JsValueRef *promise, _Out_ JsValueRef *resolve,
     });
 }
 
+CHAKRA_API JsGetPromiseStatusCode(_In_ JsValueRef promise, _Out_ JsPromiseStatusCode *statusCode)
+{
+    return ContextAPIWrapper<JSRT_MAYBE_TRUE>([&](Js::ScriptContext *scriptContext, TTDRecorder& _actionEntryPopper) -> JsErrorCode {
+        PERFORM_JSRT_TTD_RECORD_ACTION_NOT_IMPLEMENTED(scriptContext);
+
+        VALIDATE_JSREF(promise);
+        PARAM_NOT_NULL(statusCode);
+
+        *statusCode = JsPromiseStatusCode_Undefined;
+
+        if (!Js::JavascriptPromise::Is(promise))
+        {
+            return JsErrorInvalidArgument;
+        }
+
+        Js::JavascriptPromise *jsPromise = Js::JavascriptPromise::FromVar(promise);
+        Js::JavascriptPromise::PromiseStatus status = jsPromise->GetStatus();
+
+        switch (status)
```

Comment:
Should the status be undefined if jsPromise->GetResult() is null here? Just so that we don't return a good status code and then have GetPromiseResult return InvalidArgument- that could be confusing to callers

Reply:
A Promise only has a result once it's resolved or rejected, so `GetResult()` will return nullptr even in the `Unresolved` case.